ibazel/ibazel.go

```diff
@@ -99,6 +106,35 @@ func New() (*IBazel, error) {
 	return i, nil
 }
 
+func (i *IBazel) getWorkspacePath() (string, error) {
+	path, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+
+	volume := filepath.VolumeName(path)
+
+	for {
+		// filepath.Dir() includes a trailing separator if we're at the root
+		if path == volume + string(filepath.Separator) {
+			path = volume
+		}
+
+		// Check if we're at the workspace path
+		if _, err := os.Stat(filepath.Join(path, "WORKSPACE")); !os.IsNotExist(err) {
+			return path, nil
+		}
+
+		// If we've reached the root, then we know the cwd isn't within a workspace
+		if path == volume {
```

Comment:
This should fail when called from inside a test. If it doesn't then it is depending on a fragile implementation detail of Bazel's test launcher that will change in the future and break.

Reply:
Are you implying that this should just default to the cwd if a WORKSPACE isn't found (and let bazel give the error)? Or would you suggest some other way to keep this from failing in a test?

Reply:
I think this is an interface, "WORKSPACEFinder", or something and that there can be a trivial testing fake that returns $CWD or empty string or something as the path.